src/gui/MainWindow.cpp

```diff
@@ -625,6 +625,7 @@ SubWindow* MainWindow::addWindowedWidget(QWidget *w, Qt::WindowFlags windowFlags
 	SubWindow *win = new SubWindow(m_workspace->viewport(), windowFlags);
 	win->setAttribute(Qt::WA_DeleteOnClose);
 	win->setWidget(w);
+	win->resize(w->sizeHint());
```

Comment:
The default for widgets is a size hint of `(-1, -1)` (i.e. `!isValid()`). Should the code still resize in such a case?

Reply:
Hard to say. I assume `resize` won't do anything if it receives invalid size parameter, because nothing seems to be broken. But I don't see it mentioned in documentation so avoiding calling it in that case probably won't hurt..